utils/deps: always show missing deps

doronbehar opened the pull request:
When trying to use PaddleX on NixOS, which doesn't allow imperative installation of Python packages with `pip`, I found this change to be useful, and thought that other users might benefit from it. Original commit message:

> When first trying out paddlex, it can be hard to figure out how to increase the logging verbosity. If there are missing dependencies and this raises an error, seeing the exact list is helpful either way.

doronbehar committed:
utils/deps: show missing deps in error, when appropriate

When first trying out paddlex, it can be hard to figure out how to
increase the logging verbosity. If there are missing dependencies and
this raises an error, seeing the exact list is helpful.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/PaddlePaddle/PaddleX?pullRequest=4876) <br/>All committers have signed the CLA.

Bobholamovic commented:
Please merge the commits to the develop branch.

doronbehar commented:
> Please merge the commits to the develop branch.

Done.

Bobholamovic reviewed (commented):
Considering that `is_extra_available` is sometimes used only to check whether the extra is available in order to branch the logic, logging an error-level message here does not seem appropriate.

I suggest adding a parameter to `is_extra_available` to control whether an error-level log should be emitted. This flag can then be enabled in assertion-style functions such as `require_extra`, so that users can still receive helpful diagnostic information when necessary.

doronbehar commented:
OK I see this makes sense, I wasn't aware that `is_extra_available` was used in a way that shouldn't raise an error somewhere. I modified it and modified only `require_extra` to make it log an error in case a dependency is missing.
